src/AuditTrailModule/javasource/test_crm/tests/TestAuditInheritance.java

```diff
@@ -0,0 +1,226 @@
+package test_crm.tests;
+
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.Date;
+
+import com.mendix.core.Core;
+import com.mendix.core.CoreException;
+import com.mendix.systemwideinterfaces.core.IContext;
+import com.mendix.systemwideinterfaces.core.IMendixIdentifier;
+
+import org.junit.Before;
+import org.junit.Test;
+
+import audittrail.proxies.TypeOfLog;
+import system.proxies.User;
+import test_crm.proxies.Company;
+import test_crm.proxies.Group;
+import test_crm.tests.actual.ActualLog;
+import test_crm.tests.expected.ExpectedLogLineAttribute;
+import test_crm.tests.expected.ExpectedLog;
+import test_crm.tests.expected.ExpectedLogLineReferenceSet;
+
+public class TestAuditInheritance {
+	private IContext context;
+
+	private IMendixIdentifier adminId;
+	private Group group;
+	private Group group2;
+
+	@Before
+	public void beforeTesting() throws CoreException {
+		this.context = Core.createSystemContext();
+
+		this.adminId = User.load(this.context, String.format("[%1s = '%2s']", User.MemberNames.Name, Core.getConfiguration().getAdminUserName()))
+			.get(0).getMendixObject().getId();
+
+		this.group = new Group(context);
+		group.setCode(GROUP_CODE);
+		group.setName(GROUP_NAME);
+		group.commit();
+
+		this.group2 = new Group(context);
+		group2.setCode(GROUP_CODE2);
+		group2.setName(GROUP_NAME2);
+		group2.commit();
+	}
+
+	@Test
+	public void testCreateRecord() throws InterruptedException, CoreException {
+		final Date initialDate = new Date();
+		// Create new record
+		final Company company = createBaseCompany();
+
+		// Assert log was created
+		final ExpectedLog expectedLog = new ExpectedLog(4, TypeOfLog.Add, Company.entityName, adminId, initialDate, Company.entityName);
+		expectedLog.addLogLine(new ExpectedLogLineAttribute(Company.MemberNames.Name.toString(), "", NAME));
```

Comment:
I did one better (I think): the field can be any enum (since all `MemberNames` everywhere are) and static imported those enum values we need. Works just the same here and we reuse a definition generated by Studio.